scripts/manager_aura_api.lua

```diff
@@ -0,0 +1,108 @@
+-- where <sIdentifier> is the identifier used to register custom descriptors
+
+-- fCustomApplicable(nodeEffect, rSource, rTarget, rDetails)
+--    nodeEffect - DB node of the AURA effect
+--    rSource -- Actor nodeEffect is on
+--    rTarget -- Actor to determine if an applied effect should be added
+--    rDetails - Metadata about the AURA. Descriptors that match what was registered for are available in rDetails.<sIdentifier>
+--               where <sIdentifier> is the identifier used to register custom descriptors
+--     Return false if the applied aura should NOT be applied to rTarget else return true
+function registerIsApplicable(sIdentifier, fCustomApplicable)
+    if not registerHelper(sIdentifier) or not fCustomApplicable then
+        return
+    end
+    aCustomDescriptors[sIdentifier].fCustomApplicable = fCustomApplicable
+end
+
+-- Register custom descriptors that you are interested in and can be processed later. Only register
+-- non-not descriptors, the NOT version will match automaticlly. These are exact matches. If you need
+-- a non-exact match, you'll need to write your own match function and register it with AuraAPI.registerDescriptorMatch
+
+-- sIdentifier - is your custom identifier used to register custom descriptors. This can be anything except
+--               what is defined in AuraEffect.rBaseDetails
+-- sDescriptor - is a descriptor to register, or an array of descriptors to register
+function registerDescriptors(sIdentifier, sDescriptor)
+    if not registerHelper(sIdentifier) or not sDescriptor then
+        return
+    end
+
+    if type(sDescriptor) == 'table' then
+        for _, sTableDescriptor in ipairs(sDescriptor) do
+            sTableDescriptor = StringManager.trim(sTableDescriptor:lower())
+            table.insert(aCustomDescriptors[sIdentifier].aDescriptors, sTableDescriptor)
+        end
+    else
+        local sCleanDescriptor = StringManager.trim(sDescriptor:lower())
+        table.insert(aCustomDescriptors[sIdentifier].aDescriptors, sCleanDescriptor)
+    end
+end
+
+-- Internal. Used  to process custom descriptor match
+function processDescriptor(sDescriptorClean, bNot)
+    local sReturn = false
+    for sIdentifier, rCustom in pairs(aCustomDescriptors) do
+        if (StringManager.contains(rCustom.aDescriptors, sDescriptorClean) or
+                (rCustom.fCustomMatch and rCustom.fCustomMatch(sDescriptorClean, bNot))) then
+            sReturn = sIdentifier
+            break
+        end
+    end
+    return sReturn
+end
+
+
+-- Internal. Used  to process custom isApplicable
+function isAuraApplicable(nodeEffect, rSource, rTarget, rDetails)
+    local bReturn = true
+    for _, rTable in pairs(aCustomDescriptors) do
+        if rTable.fCustomApplicable and not rTable.fCustomApplicable(nodeEffect, rSource, rTarget, rDetails) then
+            bReturn = false
+            break
+        end
+    end
+    return bReturn
+end
+
+-- match on 6 char hex string
```

Comment:
isHex() needs to match on an 8 character hex string.  the FG color strings include alpha at the beginning, so its AARRGGBB.

Reply:
Nice catch. updated